app/services/platform-apps/api/modules/scene-transitions.ts

```diff
@@ -109,6 +111,80 @@ export class SceneTransitionsModule extends Module {
     throw new Error('Not Implemented');
   }
 
+  /**
+   * Get a list of transitions
+   *
+   * This includes all transitions the user has set up in SLOBS.
+   * For transitions managed by this App use `getAppTransitions`.
+   *
+   * @see {getAppTransitions}
+   */
+  @apiMethod()
+  async getTransitions(_ctx: IApiContext): Promise<Array<ITransition>> {
+    return this.transitionsService.state.transitions;
+  }
+
+  /**
+   * Get a list of transitions belonging to this App.
+   *
+   * @param ctx API context
+   * @return A list of Transition objects belonging to the current App
+   */
+  @apiMethod()
+  async getAppTransitions(ctx: IApiContext): Promise<Array<ITransition>> {
+    return Object.keys(this.transitionsService.obsTransitions)
+      .filter(obsTransitionId => {
+        const settings = this.transitionsService.getPropertiesManagerSettings(obsTransitionId);
+
+        return settings && settings.appId === ctx.app.id;
+      })
+      .map(id => this.transitionsService.getTransition(id));
+  }
+
+  /**
+   * Set a transition as the default transition
+   *
+   * @param transitionId ID of the transition to be set as default
+   * @return Whether setting the default succeeded
+   */
+  @apiMethod()
+  async setDefaultTransition(_ctx: IApiContext, transitionId: string): Promise<boolean> {
+    return tryCatchToBool(() => this.transitionsService.setDefaultTransition(transitionId));
```

Comment:
What specific case are you trying to catch here?  The API system will automatically catch any exceptions your API handler raises, and reject the promise with the exception error message.  I'm worried that by reducing this to a boolean it may be less obvious to the developer why it isn't working.

Reply:
I've removed all of these including the error handler.